Pcap++/header/PcapFilter.h

```diff
@@ -286,19 +294,46 @@ namespace pcpp
 		 * @param[in] ipAddress The IPv4 address to build the filter with. If this address is not a valid IPv4 address an error will be
 		 * written to log and parsing this filter will fail
 		 */
-		void setAddr(const std::string& ipAddress) { m_Address = ipAddress; }
+		void setAddr(const std::string& ipAddress) { this->setAddr(IPAddress(ipAddress)); }
+
+		void setAddr(const IPAddress &ipAddress)
+		{
+			if (!ipAddress.isIPv4() && hasMask())
+			{
+				// TODO: What exception type to throw here?
+				throw std::runtime_error("Attempting to set non-IPv4 address while an IPv4 mask is set. "
+										 "Please clear the mask before setting a non-IPv4 address.");
+			}
+
+			m_Address = ipAddress;
+		}
 
 		/**
 		 * Set the IPv4 mask
 		 * @param[in] ipv4Mask The mask to use. Mask should also be in a valid IPv4 format (i.e x.x.x.x), otherwise parsing this filter will fail
 		 */
-		void setMask(const std::string& ipv4Mask) { m_IPv4Mask = ipv4Mask; m_Len = 0; }
+		void setMask(const std::string& ipv4Mask)
+		{
+			if (!m_Address.isIPv4())
```

Comment:
maybe have a branch for `isIPv6`?

Reply:
What do you mean?

Reply:
I meant maybe better to also have an if branch for `m_Address.isIPv6()`

Reply:
If you are talking about the `not isIPv4` to be replaced with `isIPv6`, I have it as `not IPv4` because I think it better describes the requirement for the mask. It requires a IPv4 explicitly, so the rationale is "fail everything that is not IPv4". At the moment IPv6 is the only other IP version that is not v4, but however unlikely, that does not mean it will always be the case.

Reply:
I believed we could offer a more specific message, such as "it's an IPv6 address, not an IPv4 address." However, perhaps it's not worth the effort.